docs/source/benchmarking.md

```diff
@@ -0,0 +1,69 @@
+# Benchmarking Performance
+
+Benchmarking packet parsing is challenging because performance is greatly impacted by the complexity of the packet 
+structures being parsed. There are a few measures by which we can assess the performance of Space Packet Parser.
+
+*Note: Throughout the Space Packet Parser repo and documentation space, 
+B/kB means bytes/kilobytes and b/kb means bits/kilobits.*
+
+Common factors affecting performance:
+- Presence of calibrators and context calibrators
+- Complexity of container inheritance structure
+- Number and size of fields in a packet
+- Presence of large binary blobs (=high kbps)
+
+## Packets Per Second
+
+This is a metric we are often asked about. Unfortunately, the answer is that it depends on which packets are 
+being parsed: how many fields are in each packet and how much extra work the parser is doing to sort out complex 
+packet structures and evaluate calibrators.
+
+## Kilobits Per Second
+
+This metric is often used when discussing data volumes and downlink bandwidths to make sure that a data processing 
+system can keep up with the data rate from a spacecraft in the time allowed for processing. This number is also 
+affected by packet structures. It will be high for simple packets containing large binary blobs and low for 
+complex packets containing many small fields.
+
+## Results 
+
+As a baseline, for relatively simple packets (these are JPSS-1 spacecraft geolocation packets containing attitude 
+and ephemeris data), we benchmarked using 7200 packets with a consistent size of 71B per packet. These packets contain 
+32-bit floats and integers of various sizes.
+
+- **26405-34620 packets per second**
+- **14998-19664 kilobits per second**
+
+```
+---------------------------------------------------------- benchmark: 1 tests ---------------------------------------------------------
+Name (time in ms)                             Min       Max      Mean   StdDev    Median      IQR  Outliers     OPS  Rounds  Iterations
+---------------------------------------------------------------------------------------------------------------------------------------
+test_benchmark_simple_packet_parsing     207.9945  272.7036  243.0210  16.8787  239.5151  20.2260       7;0  4.1149      20           1
+---------------------------------------------------------------------------------------------------------------------------------------
+
+Legend:
+  Outliers: 1 Standard Deviation from Mean; 1.5 IQR (InterQuartile Range) from 1st Quartile and 3rd Quartile.
+  OPS: Operations Per Second, computed as 1 / Mean
+============================== 1 passed in 5.42s ===============================
+Progress: [====================]100% [Elapsed: 0:00:00.234986, Parsed 511200 bytes (7200 packets) at 17403kb/s (30640pkts/s)]
```

Comment:
FYI that any printed progress really slows the whole thing down too in my experience. So when profiling I'd remove all prints/logs to get your best numbers.

Reply:
Yeah, it takes this test from a mean of like 233ms to 247ms. I'll mention this in the docs.